docker-compose.yml

```diff
@@ -3,85 +3,51 @@ version: "3"
 services:
   zk:
     image: zookeeper:3.4
-    ports:
-      - 2188:2188
     environment:
       ZOO_MY_ID: 1
       ZOO_PORT: 2188
 
   master:
-    image: mesosphere/mesos-master:1.2.1
-    network_mode: host
-    pid: host
-    userns_mode: host
-    privileged: true
+    image: mesosphere/mesos-master:${MESOS_VERSION-1.2.1}
+    entrypoint:
+      - sh
+      - -c
+      - MESOS_HOSTNAME=$$(ip route | tail -n 1 | cut -d ' ' -f 12) mesos-master --registry=in_memory
     environment:
-      MESOS_ZK: zk://${DOCKER_IP}:2188/mesos
-      MESOS_QUORUM: 1
+      MESOS_ZK: zk://zk:2188/mesos
+      MESOS_QUORUM: 3
       MESOS_CLUSTER: docker-compose
-      MESOS_HOSTNAME: ${DOCKER_IP}
     depends_on:
       - zk
 
-  slave-one:
-    image: mesosphere/mesos-slave:1.2.1
-    network_mode: host
-    pid: host
-    userns_mode: host
-    privileged: true
-    environment:
-      MESOS_MASTER: zk://${DOCKER_IP}:2188/mesos
-      MESOS_CONTAINERIZERS: docker
-      MESOS_PORT: 5051
-      MESOS_RESOURCES: ports(*):[11000-11199]
-      MESOS_HOSTNAME: ${DOCKER_IP}
-      MESOS_ADVERTISE_IP: ${DOCKER_IP}
-      MESOS_WORK_DIR: /tmp/mesos
-    volumes:
-      - /sys/fs/cgroup:/sys/fs/cgroup
-      - /usr/local/bin/docker:/usr/bin/docker
-      - /var/run/docker.sock:/var/run/docker.sock
-    depends_on:
-      - zk
-
-  slave-two:
-    image: mesosphere/mesos-slave:1.2.1
-    network_mode: host
-    pid: host
-    userns_mode: host
+  slave:
+    image: mesosphere/mesos-slave:${MESOS_VERSION-1.2.1}
     privileged: true
```

Comment:
I always get `privileged mode is incompatible with user namespaces` when I run with `privileged: true`, without `userns_mode: host`

Reply:
my versions: ```tnorris-macOS:mesos-actor tnorris$ docker version
Client:
 Version:	17.12.0-ce
 API version:	1.34 (downgraded from 1.35)
 Go version:	go1.9.2
 Git commit:	c97c6d6
 Built:	Wed Dec 27 20:03:51 2017
 OS/Arch:	darwin/amd64

Server:
 Engine:
  Version:	17.11.0-ce
  API version:	1.34 (minimum version 1.12)
  Go version:	go1.8.5
  Git commit:	1caf76c
  Built:	Mon Nov 20 18:39:28 2017
  OS/Arch:	linux/amd64
  Experimental:	false
tnorris-macOS:mesos-actor tnorris$ ```

Reply:
Using docker machine on my mac:

```
Client:
 Version:       18.02.0-ce
 API version:   1.36
 Go version:    go1.9.3
 Git commit:    fc4de44
 Built: Wed Feb  7 21:13:05 2018
 OS/Arch:       darwin/amd64
 Experimental:  false
 Orchestrator:  swarm

Server:
 Engine:
  Version:      18.02.0-ce
  API version:  1.36 (minimum version 1.12)
  Go version:   go1.9.3
  Git commit:   fc4de44
  Built:        Wed Feb  7 21:20:09 2018
  OS/Arch:      linux/amd64
  Experimental: false
```

And on linux:
```
Client:
 Version:      17.05.0-ce
 API version:  1.29
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 22:10:54 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.05.0-ce
 API version:  1.29 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 22:10:54 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

This all seems to work. Do you have anything in your docker service daemon.json and why do you need namespaces on your dev box?

Reply:
oh this is because I'm using the OpenWhisk docker-machine tweaks: https://github.com/apache/incubator-openwhisk/blob/master/tools/macos/docker-machine/tweak-dockermachine.sh#L12

I forget which parts (of OpenWhisk ansible deploy) don't work without these...hrm

Reply:
Ah okay yea so nothing strictly to do with mesos there, so I'll just gloss over that detail.